2-Authorization-I/1-call-graph/SPA/src/utils/claimUtils.js

```diff
@@ -0,0 +1,213 @@
+                    changeDateFormat(claims[key]),
+                    "The exp (expiration time) claim identifies the expiration time (as UNIX timestamp) on or after which the JWT must not be accepted for processing. It's important to note that in certain circumstances, a resource may reject the token before this time. For example, if a change in authentication is required or a token revocation has been detected.",
+                    index,
+                    claimsObj
+                );
+                index++;
+                break;
+            case 'name':
+                populateClaim(
+                    key,
+                    claims[key],
+                    "The name claim provides a human-readable value that identifies the subject of the token. The value isn't guaranteed to be unique, it can be changed, and it's designed to be used only for display purposes. The profile scope is required to receive this claim.",
+                    index,
+                    claimsObj
+                );
+                index++;
+                break;
+            case 'preferred_username':
+                populateClaim(
+                    key,
+                    claims[key],
+                    'The primary username that represents the user. It could be an email address, phone number, or a generic username without a specified format. Its value is mutable and might change over time. Since it is mutable, this value must not be used to make authorization decisions. It can be used for username hints, however, and in human-readable UI as a username. The profile scope is required in order to receive this claim.',
+                    index,
+                    claimsObj
+                );
+                index++;
+                break;
+            case 'nonce':
+                populateClaim(
+                    key,
+                    claims[key],
+                    'The nonce matches the parameter included in the original /authorize request to the IDP. If it does not match, your application should reject the token.',
+                    index,
+                    claimsObj
+                );
+                index++;
+                break;
+            case 'oid':
+                populateClaim(
+                    key,
+                    claims[key],
+                    'The oid (user’s object id) is the only claim that should be used to uniquely identify a user in an Azure AD tenant. The token might have one or more of the following claim, that might seem like a unique identifier, but is not and should not be used as such.',
+                    index,
+                    claimsObj
+                );
+                index++;
+                break;
+            case 'tid':
+                populateClaim(
+                    key,
+                    claims[key],
+                    'The tenant ID. You will use this claim to ensure that only users from the current Azure AD tenant can access this app.',
```

Comment:
>The tenant ID [](http://example.com/codeflow?start=21&length=13)

The id of the tenant where this app resides

Reply:
Updated!